web-app/js/portal/data/LayerSelectionModel.js

```diff
@@ -57,7 +57,11 @@ Portal.data.LayerSelectionModel = Ext.extend(Ext.util.Observable, {
 
     // TODO: unit tests?
     _linkToOpenLayer: function(layerLink, dataCollection) {
-        var layerDisplayName = dataCollection.get('title');
+
+        var layerDisplayName;
+        if (layerLink.title == "") {
+            layerDisplayName = dataCollection.get('metadataRecord').get('title');
```

Comment:
Can be `layerDisplayName = dataCollection.getTitle();` I think.  Better, because we're not accessing the internals of `DataCollection`.